db/query/configure.sql

```diff
@@ -0,0 +1 @@
+CREATE EXTENSION IF NOT EXISTS postgis;
```

Comment:
What do you think it would take to put this in a migration? I really feel like not including this there could create headaches for us in the future. The point of migrations is to track changes to the database that power this API and be as deterministic as possible. That should include installing extensions that subsequent migrations depend on.

I know Drizzle makes it tricky to do something retroactively like this (as would any ORM with migrations, not a dig at Drizzle) but I don't think it would be that problematic to "restart" the migrations timeline for this.

Reply:
I don't feel comfortable restarting the migrations for this. We should build cultural habits of respecting the integrity of the shared environments- if we restart the migrations then any outstanding instances will also need to be wiped and rebuilt. We should reserve that option as a last resort. Otherwise, we risk normalizing it.

Also, I mentally model postgis installation as closer to "database setup" than "database migration". We already accept the database needs to be in some minimally configured state before we run migrations against it. For example, we could technically initialize the zoning database with [a create database sql command](https://www.postgresql.org/docs/current/sql-createdatabase.html). However, we place this step in the "setup" phase because it practically makes more sense. The fact we're dealing with two different postgis setup pathways right now is itself evidence that it makes sense tailor the setup process based on the environment. The migrations shouldn't need to be aware of this environment.

Finally, I don't really see postgis installation creating headaches as a separate step. We already use distinct commands to handle different application setup steps. Local application setup is at least:

```
docker compose up --build
nvm use
npm i
npm run db:migrate
npm run db:load
npm run build
npm run start
```

With postgis installation as a separate step, the process is:
```
docker compose up --build
nvm use
npm i
npm run db:configure
npm run db:migrate
npm run db:load
npm run build
npm run start
```
For production releases, these steps will be codified in GitHub actions to ensure they're applied consistently.

If the concern is for developers forgetting to run `configure` before they run `migrate`, the developers should be able to recover quickly. The migrations will fail loudly because they don't have access to the specified spatial types. They will also fail safely because they're in transactions. Also, we can chain the whole setup with a single npm script like:

```"db:setup": "npm run db:configure && npm run db:migrate && npm run db:copy"```

Reply:
I think we just disagree (which is OK!) about where to draw the line of what "migrations" are responsible for, though I definitely take your point that it feels arbitrary to include `CREATE EXTENSION` calls but not `CREATE DATABASE` ones. I'm OK with moving forward with your approach.

Just to explain where I'm coming from, my mental model of migrations is more that they should try to include all of the SQL that has to run for the database to be used by the application. I'm ok with having that `load` step because that is essentially our implementation of "seeding", which is typically considered separate from migrations by most ORMs that do have a mechanism for it, but I just want to make sure we don't end up with a bunch of `db:something`s down the road and the Drizzle migrations become less and less deterministic.

One last thought - if we do implement something like that `db:setup` script, we could leverage NPM's [pre and post scripts feature](https://docs.npmjs.com/cli/v9/using-npm/scripts#pre--post-scripts) if we want. Just thought - some folks avoid using that feature because it feels hand wave-y.